### PR TITLE
Ignore combat logs for AI

### PR DESCRIPTION
**Pull request by psykzz**
This is an attempt to allows admins to read combat logs on WO.

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -398,6 +398,10 @@ Proc for attack log creation, because really why not
 */
 
 /proc/log_combat(atom/user, atom/target, what_done, atom/object, addition)
+	var/eitherIsAI = user.GetComponent(/datum/component/ai_controller) || target.GetComponent(/datum/component/ai_controller)
```
Comment (spookydonut) [inline code/game/atoms.dm:401]: GetComponent is slow, i wouldnt do this imo

Comment (psykzz) [reply code/game/atoms.dm:401]: Do we want to just have the component update a flag on the mob to say its an AI mob instead maybe?

Comment (psykzz) [reply code/game/atoms.dm:401]: Could be `flags_atom & AI_CONTROLLED` or something

Comment (spookydonut) [reply code/game/atoms.dm:401]: maybe just SEND_SIGNAL(user+target COMSIG_LOG_COMBAT) & DONTLOG or something

**Comment by psykzz** [comment]
Updated to signals 
image

File: code/game/atoms.dm [hunk 2]
```diff
@@ -398,6 +398,11 @@ Proc for attack log creation, because really why not
 */
 
 /proc/log_combat(atom/user, atom/target, what_done, atom/object, addition)
+	var/signalOut = SEND_SIGNAL(user, COMSIG_COMBAT_LOG)
+	signalOut |= SEND_SIGNAL(target, COMSIG_COMBAT_LOG)
+	if (signalOut & DONT_LOG)
```
Comment (spookydonut) [inline code/game/atoms.dm:403]:
```suggestion
	if (SEND_SIGNAL(user, COMSIG_COMBAT_LOG) | SEND_SIGNAL(target, COMSIG_COMBAT_LOG) & DONT_LOG)
```

**Pull request merged**